modules/geolocation/unittests.py

```diff
@@ -121,6 +121,89 @@ def test_camera_offset_from_origin_pointing_sideways_with_not_enough_downward_pi
         np.testing.assert_equal(actual, expected)
 
 
+class TestGetNonCollinearPoints(unittest.TestCase):
+    """
+    Tests get_non_collinear_points()
+    """
+    def setUp(self):
+        self.locator = geolocation.Geolocation()
+        return
+    
+    def test_empty_input_array(self):
+        # Setup
+        coordinatesArray = np.array([])
+        expected = np.zeros(shape=(4, 2))
+        
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    def less_than_four_points_collinear(self):
+        # Setup
+        coordinatesArray = np.array([[0, 0], [1, 1], [-1, -1]])
+        expected = np.empty(shape=(4, 2))
+
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    def less_than_four_points_non_collinear(self):
+        # Setup
+        coordinatesArray = np.array([[0, 0], [-1, 1], [12, 7]])
+        expected = np.empty(shape=(4, 2))
+
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    def test_four_points_collinear(self):
+        coordinatesArray = np.array([[0, 0], [1, 1], [2, 2], [10, 10]])
+        expected = np.empty(shape=(4, 2))
+
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    def test_four_points_non_collinear(self):
+        coordinatesArray = np.array([[0, 0], [10, 12], [-1, -1], [100, 0]])
+        expected = np.array([[0, 0], [10, 12], [-1, -1], [100, 0]])
+
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    def test_more_than_four_points_one_correct_case(self):
+        coordinatesArray = np.array([[0, 0], [10, 12], [-1, -1], [100, 0], [4, 4]])
+        expected = np.array([[0, 0], [10, 12], [-1, -1], [100, 0]])
+
+        # Run
+        actual = self.locator.get_non_collinear_points(coordinatesArray)
+
+        # Test
+        np.testing.assert_almost_equal(actual, expected)
+
+    
+    def test_more_than_four_points_no_correct_cases(self):
+        coordinatesArray = np.array([[0, 0], [1, 1], [2, 2], [10, 10], [15, 15], [0, 0]])
```

Comment:
Add one point outside of this line of points to make things slightly more interesting.

Reply:
Also, there is a duplicate [0, 0], why is that?